### PR TITLE
remote: pull refactor to match default behavior of cgit

### PR DESCRIPTION
**Pull request by mcuadros**
This PR implement the default behavior of cgit, until now Pull was wrongly implemented changing the branch instead of merging the incoming branch into the current HEAD.

Fixes https://github.com/src-d/go-git/issues/470

### CODE REVIEW

File: COMPATIBILITY.md [hunk 1]
```diff
@@ -27,7 +27,7 @@ is supported by go-git.
 | tag                                   | ✔ |
 | **sharing and updating projects** |
 | fetch                                 | ✔ |
-| pull                                  | ✔ |
+| pull                                  | ✔ | Only supports merges where the merge can be resolved as a fast-forward. |
```
Comment (smola) [inline COMPATIBILITY.md]: Maybe reword as `Only supports fast-forward merges`

File: worktree.go [hunk 2]
```diff
@@ -36,6 +37,8 @@ type Worktree struct {
 // Pull incorporates changes from a remote repository into the current branch.
 // Returns nil if the operation is successful, NoErrAlreadyUpToDate if there are
 // no changes to be fetched, or an error.
+//
+// Pull only supports merges where the can be resolved as a fast-forward.
```
Comment (smola) [inline worktree.go]: Maybe reword, see previous comment.

File: worktree.go [hunk 3]
```diff
@@ -44,6 +47,8 @@ func (w *Worktree) Pull(o *PullOptions) error {
 // branch. Returns nil if the operation is successful, NoErrAlreadyUpToDate if
 // there are no changes to be fetched, or an error.
 //
+// Pull only supports merges where the can be resolved as a fast-forward.
```
Comment (smola) [inline worktree.go]: Same here.

File: COMPATIBILITY.md (hunk 1 above)
Comment (mcuadros) [inline COMPATIBILITY.md]: is copy from the git documentation
